src/Modal.js

```diff
@@ -233,6 +230,7 @@ class Modal extends React.Component {
 
   componentDidMount() {
     this._isMounted = true;
+    this.manager = new ModalManager();
```

Comment:
- You should only instantiate this if a manager wasn't specified via `props`.
- By default, we still want a single global singleton manager. We just want to delay when it gets initialized.

So I think I'd do something like:

```js
let manager; // Maybe export this?

class Modal {
  // ...

  getModalManager() {
    if (this.props.manager) {
      return this.props.manager;
    }

    if (!manager) {
      manager = new ModalManager();
    }

    return manager;
  }

  // ...
}
```

Reply:
My bad, I previously wrote that as `this.manager = this.props.manager || new ModalManager` to still allow injection from props, removed when testing some issue and forgot to revert. Thanks catching 👍

Reply:
But even so that doesn't quite preserve the semantics, as it means that changing the passed-in modal manager via the props would have no effect.

Reply:
I saw your point, you want to keep a singleton of ModalManager instance which keeps global knowledge of `this.modals`.

That might be better wrapped in a separate module I guess, maybe something like this?

```
// Modal.js
import { getModalManager } from './ModalManager';

//.....
  componentDidMount() {
    this._isMounted = true;
    this.manager = this.props.manager || getModalManager();
```

And

```
// ModalManager.js
// ....

let manager;

function getModalManager() {
  if (manager) {
     return manager;
  }
 manager = new ModalManager();
 return manager;
}
```

Sounds good?

BTW maybe similar needs to be done in react-bootstrap later too https://github.com/react-bootstrap/react-bootstrap/blob/0bb8a6fd4ad110c49e5e89f05228ade0968652bf/src/Modal.js#L154

Reply:
See the example code in https://github.com/react-bootstrap/react-overlays/pull/297#discussion_r294290387. Do that, then use `this.getModalManager()`. The issue with your proposed solution above is that it wouldn't respect changes to the prop.